readconf.c

```diff
@@ -1734,18 +1738,140 @@ read_config_file_depth(const char *filename, struct passwd *pw,
 
 	if ((f = fopen(filename, "r")) == NULL)
 		return 0;
+	if (flags & SSHCONF_CHECKPERM) {		
+#if WINDOWS
+		/*		
+		file permissin are designed differently on windows
+		implementation on windows to make sure the config file is owned by the user of calling process
+		and nobody else except Administrators group and SYSTEM account has the write permission
+		*/
+		HANDLE h;
+		PSID owner_sid = NULL, user_sid = NULL;
+		PACL dacl = NULL;		
+		DWORD ret;		
+		PSECURITY_DESCRIPTOR pSD = NULL;		
+		char buf[2048];
+		int return_error = 0;
+		BOOL others_has_write_permission = FALSE;
+		
+		/*Get the sid of the calling process.*/
+		if ((user_sid = getusid()) == NULL) {
+			snprintf(buf, sizeof(buf), "failed to retrieve the user sid of the calling process with error: %s", strerror(errno));
+			return_error = -1;
```

Comment:
assign return_error = -1 at start, and =0 before succeeding. Reduces a line of code for each failure case